cpp/lib/Authenticator.cpp

```diff
@@ -116,29 +128,27 @@ extern "C" {
     }
     catch (...)
     {
-      SET_SNOWFLAKE_ERROR(&conn->error, SF_STATUS_ERROR_GENERAL,
-                          "authentication failed",
-                          SF_SQLSTATE_GENERAL_ERROR);
       return SF_STATUS_ERROR_GENERAL;
     }
 
     return SF_STATUS_SUCCESS;
   }
 
-  void auth_update_json_body(SF_CONNECT * conn, cJSON* body)
+  void auth_update_json_body(SF_CONNECT *conn, cJSON* body)
   {
-    if (AUTH_OAUTH == getAuthenticatorType(conn->authenticator)) 
+    cJSON* data = snowflake_cJSON_GetObjectItem(body, "data");
+    if (!data)
     {
-        cJSON* data = snowflake_cJSON_GetObjectItem(body, "data");
-        if (!data)
-        {
-            data = snowflake_cJSON_CreateObject();
-            snowflake_cJSON_AddItemToObject(body, "data", data);
-        }
+          data = snowflake_cJSON_CreateObject();
+          snowflake_cJSON_AddItemToObject(body, "data", data);
+    }
+
+    snowflake_cJSON_DeleteItemFromObject(data, "AUTHENTICATOR");
```

Comment:
auth_update_json_body is a C wrapper for updateDataMap in C++. AUTH_OAUTH is an exception since it's quite simple so we don't have authenticator implementation for it in C++. For others we should fully rely on updateDataMap in case any common logic here might not apply.

Reply:
I moved this to connection.c